From 7b8c94d9f7f4fb2401a18347c15314ca724e6b73 Mon Sep 17 00:00:00 2001
From: Chris Lamb <lamby@debian.org>
Date: Fri, 24 Jul 2009 11:37:20 +0200
Subject: [PATCH] Escape "-" to ensure interpretation as minus sign, not hyphen

"-" chars are interpreted as hyphens (U+2010) by groff, not as minus
signs (U+002D). Since options to programs use minus signs (U+002D), this
means for example in UTF-8 locales that you cannot cut&paste options,
nor search for them easily.

"-" must be escaped ("\-") to be interpreted as minus. If we really
intend a hyphen, we can write it as "\(hy" to emphasise that fact. See
groff(7) and especially groff_char(7) for details.

It might be possible to fix ths by jumping into a different context,
but I don't know enough manpage stuff.

Signed-off-by: Chris Lamb <lamby@debian.org>
---
 preload/run/fiu-run.1 |    2 +-
 utils/fiu-ctrl.1      |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/preload/run/fiu-run.1 b/preload/run/fiu-run.1
index bcc9788..888cce4 100644
--- a/preload/run/fiu-run.1
+++ b/preload/run/fiu-run.1
@@ -62,7 +62,7 @@ enable failure points in the POSIX and libc functions):
 
 .RS
 .nf
-fiu-run -x -e posix/io/* -p 10 -e libc/mm/malloc -p 5 fortune
+fiu\-run \-x \-e posix/io/* \-p 10 \-e libc/mm/malloc \-p 5 fortune
 .fi
 .RE
 
diff --git a/utils/fiu-ctrl.1 b/utils/fiu-ctrl.1
index ceafd6b..d739f9a 100644
--- a/utils/fiu-ctrl.1
+++ b/utils/fiu-ctrl.1
@@ -49,7 +49,7 @@ failure point \fIlibc/mm/malloc\fR with a 5% of probability to fail:
 
 .RS
 .nf
-fiu-ctrl -e posix/io/read -p 25 -e libc/mm/malloc -p 5 12345
+fiu\-ctrl \-e posix/io/read \-p 25 \-e libc/mm/malloc \-p 5 12345
 .fi
 .RE
 
@@ -58,7 +58,7 @@ failure point \fIposix/io/read\fR:
 
 .RS
 .nf
-fiu-ctrl -d posix/io/read 12345
+fiu\-ctrl \-d posix/io/read 12345
 .fi
 .RE
 
-- 
1.6.3.3

